tests/handlers/test_federation.py

```diff
@@ -190,6 +194,134 @@ def test_rejected_state_event_state(self):
 
         self.assertEqual(sg, sg2)
 
+    def test_backfill_floating_outlier_membership_auth(self):
+        """
+        As the remote federated homeserver, check that we can properly process
+        an event with auth_events that include a floating membership event from
+        the OTHER_SERVER.
+
+        Regression test, see #10439.
+        """
+
+        def filter_auth_events_on_event(event: EventBase, auth_events: List[EventBase]):
+            # Create a StateMap[str]
+            auth_event_state_map = {
+                (e.type, e.state_key): e.event_id for e in auth_events
+            }
+            # Actually strip down and use the necessary auth events
+            auth_event_ids = self._event_auth_handler.compute_auth_events(
+                event=event,
+                current_state_ids=auth_event_state_map,
+                for_verification=False,
+            )
+
+            # Replace the auth_events with the stripped down ones
+            event.auth_events = auth_event_ids
```

Comment:
You shouldn't be able to edit the auth events directly (as it invalidates all the hashes), so I am loathed to rely on this.

Instead, can we use an event builder to create the event? Something like:

```python
builder = hs.get_event_builder_factory().for_room_version(room_version, event_dict)
event = await builder.build(
    prev_event_ids=[fake_prev_event_id],
    auth_events=compute_auth_events(builder, state),
    depth=depth,
)
```

Reply:
I've switched to the event builder but I still have to edit the `signatures` to make them appear from the `OTHER_SERVER`. Otherwise they are rejected as not being signed by that server.